samcli/commands/build/command.py

```diff
@@ -128,6 +129,16 @@
     help="Path to environment variable json file (e.g., env_vars.json) to pass into build containers",
     cls=ContainerOptions,
 )
+@click.option(
+    "--container-dir-mount",
```

Comment:
For this option name and also the name in the build.toml file I will briefly consult our UX team on it! dir-mount doesn't sound exactly intuitive to me. Sorry but naming such options requires some extra attention!